File: examples/scala/src/test/scala/WhenStage.scala
import org.scalatest._
import com.tngtech.jgiven.annotation._
import com.tngtech.jgiven._

class WhenStage {

  @ExpectedScenarioState
  var someInt = 0

  @ExpectedScenarioState
  var anotherInt = 0
  
  @ProvidedScenarioState
  var result = 0
  
  def adding_the_values = {
    result = someInt + anotherInt
  }
}

File: jgiven-junit6/src/test/java/com/tngtech/jgiven/junit6/test/WhenStage.java
package com.tngtech.jgiven.junit5.test;

import com.tngtech.jgiven.annotation.ExpectedScenarioState;
import com.tngtech.jgiven.annotation.ProvidedScenarioState;
import org.junit.jupiter.api.Assertions;

public class WhenStage {

    @ExpectedScenarioState
    String someState;

    @ProvidedScenarioState
    String someResult;

    void some_action() {
        Assertions.assertNotNull(someState);
        someResult = "Some Result";
    }

    void some_action_with_a_parameter(String s) {
        someResult = "s";
    }

    public WhenStage some_failing_step() {
        Assertions.assertTrue(false, "Intentionally failing");
        return this;
    }
}
